server_manager/www/app.ts

```diff
@@ -1035,31 +1038,99 @@ export class App {
+      const bandwidthUsageHeap = new Heap(bandwidthUsageComparator);
+
+      let tunnelTimeTotal = 0;
+      const tunnelTimeComparator: Comparator<server_model.ServerMetrics> = (
+        server1,
+        server2
+      ) => server2.tunnelTime.seconds - server1.tunnelTime.seconds;
+      const tunnelTimeHeap = new Heap(tunnelTimeComparator);
+
+      let devicesTotal = 0;
+      const devicesComparator: Comparator<server_model.ServerMetrics> = (
+        server1,
+        server2
+      ) => server2.devices - server1.devices;
+      const devicesHeap = new Heap(devicesComparator);
+
+      for (const server of serverMetrics.server) {
+        bandwidthUsageTotal += server.dataTransferred.bytes;
+        bandwidthUsageHeap.push(server);
+
+        tunnelTimeTotal += server.tunnelTime.seconds;
+        tunnelTimeHeap.push(server);
+
+        devicesTotal += server.devices;
+        devicesHeap.push(server);
       }
 
-      serverView.totalUserHours = totalUserHours;
-      serverView.totalAverageDevices = totalAverageDevices;
+      // support legacy metrics view
+      serverView.totalInboundBytes = bandwidthUsageTotal;
 
-      let totalInboundBytes = 0;
-      for (const {dataTransferred} of serverMetrics.accessKeys) {
-        if (!dataTransferred) continue;
-
-        totalInboundBytes += dataTransferred.bytes;
-      }
-
-      serverView.totalInboundBytes = totalInboundBytes;
+      const NUMBER_OF_ASES_TO_SHOW = 4;
+      serverView.bandwidthUsageTotal = this.formatGigabyteValueAndUnit(
+        bandwidthUsageTotal / BYTES_IN_GIGABYTES
+      );
+      serverView.bandwidthUsageRegions = bandwidthUsageHeap
+        .top(NUMBER_OF_ASES_TO_SHOW)
+        .reverse()
+        .map(server => ({
+          title: server.asOrg,
+          subtitle: `${server.asn}AS`,
+          icon: this.countryCodeToEmoji(server.location),
+          highlight: this.formatGigabyteValueAndUnit(
+            server.dataTransferred.bytes / BYTES_IN_GIGABYTES
+          ),
+        }));
+
+      serverView.tunnelTimeTotal = this.formatHourValue(
+        tunnelTimeTotal / SECONDS_IN_HOUR
+      );
+      serverView.tunnelTimeTotalLabel = this.formatHourUnits(
+        tunnelTimeTotal / SECONDS_IN_HOUR
+      );
+      serverView.tunnelTimeRegions = tunnelTimeHeap
+        .top(NUMBER_OF_ASES_TO_SHOW)
+        .reverse()
+        .map(server => ({
+          title: server.asOrg,
+          subtitle: `${server.asn}AS`,
+          icon: this.countryCodeToEmoji(server.location),
+          highlight: this.formatHourValueAndUnit(
+            server.tunnelTime.seconds / SECONDS_IN_HOUR
+          ),
+        }));
+
+      const NUMBER_OF_DEVICES_SIGNIFICANT_DIGITS = 3;
+      serverView.devicesTotal = devicesTotal.toFixed(
+        NUMBER_OF_DEVICES_SIGNIFICANT_DIGITS
+      );
+      serverView.devicesRegions = devicesHeap
+        .top(NUMBER_OF_ASES_TO_SHOW)
+        .reverse()
+        .map(server => ({
+          title: server.asOrg,
```

Comment:
You don't always have asOrg. What's the plan when it's absent?
I usually use `${asOrg} (AS${asn})` with fallback to `AS${asn}`